### DISCUSSION TITLE
How to used changedKeys?

### DISCUSSION THREAD

**Discussion by chiester** [vazco/uniforms#1056] [Q&A] [answered]
I'm not clear on how to use changedKeys? I have a fairly basic form but in some cases, I want to know which keys in the model were changed and conditionally do different things. None of my fields are custom components, so it doesn't seem like using the hooks is an option. 

Might be thinking of `changed` instead of `changedKeys`

**Comment by radekmie** [comment, marked as answer]
Hi @chiester. What do you mean by using `changed` instead of `changedKeys`?
* `changed` is a context property, telling whether or not any of the fields have changed ([docs](https://uniforms.tools/docs/api-context-data/#available-context-data)).
* `changedKeys` is a helper ([docs](https://uniforms.tools/docs/api-helpers/#changedkeys)), used internally to calculate `changedMap` ([docs](https://uniforms.tools/docs/api-context-data/#available-context-data)).

I guess that you may want to use `changedMap` instead. It's how uniforms calculate the `changed` property of every field (`useField` call). A field is changed if `!!get(changedMap, name)` (`get` is a Lodash function).
